[DOCS]  Create a New Get Support Topic (#8849)
great-expectations/great_expectations: docs/docusaurus/docs/get_support.md

---
sidebar_label: 'Get support'
title: 'Get support'

> description: Learn what resources are avialable to help you resolve issues with GX Core.

Reviewer: ```suggestion
description: Learn what resources are available to help you resolve issues with GX Core.
```

> description: Learn what resources are available to help you resolve issues with GX Core.

Reviewer 2: ```suggestion
description: Learn what resources are available to help you resolve issues with GX OSS.
```

Reviewer 2: Found one more Core -> OSS

---

> GX Core is open source software (OSS) and relies on an active user community for its continued success. Input from users like you help GX and the community build its collective knowledge and improve GX Core. Knowing how and where to seek support will help you get the answers you need to resolve issues.

Reviewer 2: ```suggestion
GX is open source software (OSS) and relies on an active user community for its continued success. Input from users like you helps Great Expectations and the community build its collective knowledge and improve GX OSS. Knowing how and where to seek support will help you get the answers you need to resolve issues.
```

> GX Core is open source software (OSS) and relies on an active user community for its continued success. Input from users like you help GX and the community build its collective knowledge and improve GX Core. Knowing how and where to seek support will help you get the answers you need to resolve issues quickly.

Reviewer 2: "Knowing how and where to seek support will help you get the answers you need to resolve issues quickly."

We can't guarantee users will be able to resolve their issues or even get an answer (helpful or not) via our community forums or support channels. Do you think we need a softer wording to that effect here? (FWIW I tried coming up with a rewrite, but couldn't figure out how to rephrase it well).

Author: @rachhouse I removed _quickly_. It's a start!

Reviewer 2: 👍

## Try solving your issue first

The following are some recommendations for resolving issues on your own. Resolving issues on your own builds your own knowledge and provides you with an opportunity to share that knowledge with the community. 

### Search the GX documentation

> Search the docs you're using currently for an answer to your issue or question. If you're new to GX Core, try the [Quickstart](tutorials/quickstart/quickstart.md) or one of the specific getting started guides such as [Get started with Great Expectations and Databricks](tutorials/getting_started/how_to_use_great_expectations_in_databricks.md), or [Get started with Great Expectations and SQL](tutorials/getting_started/how_to_use_great_expectations_with_sql.md).  

Reviewer 2: ```suggestion
Search the docs you're using currently for an answer to your issue or question. If you're new to GX OSS, try the [Quickstart](tutorials/quickstart/quickstart.md) or one of the specific getting started guides such as [Get started with Great Expectations and Databricks](tutorials/getting_started/how_to_use_great_expectations_in_databricks.md), or [Get started with Great Expectations and SQL](tutorials/getting_started/how_to_use_great_expectations_with_sql.md).  
```

### Search with your favorite browser

> It's very likely that someone, somewhere has previously encountered or solved the issue you're currently experiencing. Perform a keyword search in your favorite browser to discover resources outside GX documentation. 

Reviewer 2: Should we mention Discourse here directly?
"discover resources outside GX documentation, such as our [Discourse forums](https://discourse.greatexpectations.io/)."

Author: @rachhouse I was going to add your edit, but I think it's best as is. A search might not locate the Discourse forum.  I know because I tried.

### Try it out

> If you're wondering what might happen when you implement a solution, try it out! If you attempt a solution in your development environment, users are unaffected and unaware of what you're up to. Sneaky. 

Reviewer 2: Sneaky 😁

## Ask the right questions and provide the right information

> Taking the time to formulate your questions properly and providing all the relevant background information can save time and improve response times. The following are some recommendations that will help the GX Developer Advocates (DAs) help you. 

Reviewer 2: "...recommendations that will help the GX Developer Advocates (DAs) and other community members help you."

Author: @rachhouse implemented as recommended.

### Include all the relevant information

> Tell the GX DAs exactly what issue you're experiencing, or the problem you're trying to solve. When asking questions, you should provide the following:

Reviewer 2: I'd recommend: "Clearly explain the issue you're experiencing, ...." since it's not always going to be DAs who respond.

Author: @rachhouse the edit was implemented as requested.

- The complete error log or logs.

- The relevant code you're trying to run.

> - The version of GX Core you're using.

Reviewer 2: ```suggestion
- The version of GX OSS you're using.
```

- A detailed explanation of what you're trying to resolve including what attempts you've made on your own to date.

### Don't post in multiple channels and use threads

When posting questions or asking for help in one of the GX Slack channels, don't post in multiple channels. Instead, select a single, relevant channel and post your message there. When responding to a post, use threads. If you need to post large code blocks, post them in a thread to avoid overwhelming the channel.

> ### Don't tag single or multiple users

Reviewer 2: ```suggestion
### Don't tag individual users
```

> If you've developed a rapport with a specific GX DA, or a community member, don't tag them in Slack posts. You never know who is available or who has the knowledge to solve your issue. Tagging specific users can limit the scope of the response. Also, the community is typically very responsive and helpful. If you don't get a response within a reasonable time, review your post and make sure you've framed your questions correctly and provided the necessary background information. If you don't get a response, or it's unhelpful, try an alternate medium to resolve your issue.

Reviewer 2: Should this read "Even you've previously developed a rapport with a specific GX DA, or a community member, don't tag them in new Slack posts."?

Author: @rachhouse the edit was implemented as requested.

> Even if you've previously developed a rapport with a specific GX DA, or a community member, don't tag them in new Slack posts. You never know who is available or who has the knowledge to solve your issue. Tagging specific users can limit the scope of the response. Also, the community is typically very responsive and helpful. If you don't get a response within a reasonable time, review your post and make sure you've framed your questions correctly and provided the necessary background information. If you don't get a response, or it's unhelpful, try an alternate medium to resolve your issue.

Reviewer 2: ```suggestion
Even if you've previously developed a rapport with a specific GX DA, or a community member, don't tag them in new Slack posts. GX DA and community member availability varies. The GX DAs operate on a rotation schedule to ensure equal time working with the open source community, and you never know who is available or who has the knowledge to solve your issue. Tagging specific users can limit the scope of the response. Also, the community is typically very responsive and helpful. If you don't get a response within a reasonable time, review your post and ensure you've framed your questions correctly and provided the necessary background information. If you don't get a response, or it's unhelpful, try an alternate medium to resolve your issue.
```

> ## Select the correct medium for questions and support

Reviewer 2: Nice categorizations below, thanks!

Author: @rachhouse thanks!

> Even if you've previously developed a rapport with a specific GX DA, or a community member, don't tag them in new Slack posts. GX DA and community member availability varies. The GX DAs operate on a rotation schedule to ensure equal time working with the open source community, and you never know who is available or who has the knowledge to solve your issue. Tagging specific users can limit the scope of the response. Also, the community is typically very responsive and helpful. If you don't get a response within a reasonable time, review your post and ensure you've framed your questions correctly and provided the necessary background information. If you don't get a response, or it's unhelpful, try an alternate medium to resolve your issue.

Author: ```suggestion
Even if you've previously developed a rapport with a specific GX DA, or a community member, don't tag them in new Slack posts. GX DA and community member availability varies. Also, the GX DAs operate on a rotation schedule to ensure equal time working with the open source community, and you never know who is available or who has the knowledge to solve your issue. Tagging specific users can limit the scope of the response. Also, the community is typically very responsive and helpful. If you don't get a response within a reasonable time, review your post and ensure you've framed your questions correctly and provided the necessary background information. If you don't get a response, or it's unhelpful, try an alternate medium to resolve your issue.
```

GX provides the following resources for questions and support:

> - If you've encountered an error and are stuck, get support on the [GX Discourse forum](https://discourse.greatexpectations.io/c/oss-support/11)

Reviewer 3: Let's say something here about how we're prioritizing questions in Discourse.

Author: @joshzzheng can you clarify your request? How are requests being prioritized?

Reviewer 3: I suggest something like "GX is prioritizing questions in Discourse (over Slack) since the answers are not lost after 90 days. We recommend this channel for quicker response."
